File: src/authent8/core/connectivity.py
"""Connectivity checks for scan runtime decisions."""

import socket


def has_internet(timeout: float = 2.5) -> bool:
    """Fast internet check using a DNS endpoint."""
    try:
        socket.create_connection(("1.1.1.1", 53), timeout=timeout)
        return True
    except OSError:
        return False

File: internal/net/connectivity.go
// Package net provides network utility functions.
package net

import (
	"net"
	"time"
)

// HasInternet performs a fast connectivity check by dialing Cloudflare DNS.
func HasInternet(timeout time.Duration) bool {
	if timeout == 0 {
		timeout = 2500 * time.Millisecond
	}
	conn, err := net.DialTimeout("tcp", "1.1.1.1:53", timeout)
	if err != nil {
		return false
	}
	conn.Close()
	return true
}
